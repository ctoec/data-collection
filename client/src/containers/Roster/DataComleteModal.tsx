import { Button, Modal } from '@ctoec/component-library';
import React from 'react';

type DataCompleteModalProps = {
  isOpen: boolean;
  closeModal: () => void;
};

export const DataCompleteModal: React.FC<DataCompleteModalProps> = ({
  isOpen,
  closeModal,
}) => {
  return (
    <Modal
      isOpen={isOpen}
      onModalClose={closeModal}
      onXClick={closeModal}
      header={<h2>We've let OEC know your data is complete!</h2>}
      content={
        <div>
          <p>
            Going forward, update your roster with new enrollments or
            withdrawals, as well as age group, funding, and site changes.
          </p>
          <p>
            We're working on a feature to allow updating from excel files. Until
            then, you can make changes directly in your ECE Reporter roster.
          </p>
          <h3>How was this experience?</h3>
          <div>
            Take a quick survey to let us know how we can make this tool easier
            to use and more valuable to childcare providers like you.
          </div>
          <div className="margin-top-4">
            <Button
              text="Back to roster"
              appearance="outline"
              onClick={() => closeModal()}
            />
            <a
              className="usa-button usa-button--default"
              href="https://forms.monday.com/forms/f5bcf5d3d611b99f45ffb8187cd6c2b9?r=use1"
              target="_blank"
            >
              Give feedback
            </a>
          </div>
        </div>
      }
    ></Modal>
  );
};
