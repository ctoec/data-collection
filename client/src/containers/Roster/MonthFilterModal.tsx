import React, { useState } from 'react';
import moment, { Moment } from 'moment';
import cx from 'classnames';
import {
  Button,
  DateInput,
  FormStatusProps,
  Modal,
} from '@ctoec/component-library';

type MonthFilterModal = {
  filterByMonth?: Moment;
  setFilterByMonth: (_: any) => void;
  toggleModal: () => void;
  isModalOpen: boolean;
};

export const MonthFilterModal: React.FC<MonthFilterModal> = ({
  filterByMonth,
  setFilterByMonth,
  toggleModal,
  isModalOpen,
}) => {
  // Duplicate of month value on roster-- needed bc we want to preserve the value but don't want to trigger the filtering until the user clicks the button
  const [selectedMonth, setSelectedMonth] = useState<Moment | undefined>(
    filterByMonth
  );
  const monthRecentEnough = selectedMonth?.isSameOrAfter(
    moment('July 2020', 'MMMM YYYY')
  );
  const monthInPast = selectedMonth?.isSameOrBefore(moment());
  let monthStatus: FormStatusProps | undefined = undefined;
  if (selectedMonth && !monthRecentEnough) {
    // Must be same or after July 2020
    monthStatus = {
      type: 'error',
      id: 'month-too-far-back',
      message: 'Month cannot be before July 2020',
    };
  } else if (selectedMonth && !monthInPast) {
    // Must be same or before right now
    monthStatus = {
      type: 'error',
      id: 'month-in-future',
      message: 'Month cannot be in the future',
    };
  }

  return (
    <Modal
      isOpen={isModalOpen}
      onModalClose={toggleModal}
      header={<h2>View past enrollments</h2>}
      content={
        <>
          <div className={cx('grid-row', { 'margin-left-3': !!monthStatus })}>
            <DateInput
              defaultValue={selectedMonth}
              onChange={(moment) => setSelectedMonth(moment)}
              label="View your roster for a past month"
              id="roster-filter-date-input"
              hideField={{
                day: true,
                calendar: true,
              }}
              status={monthStatus}
            />
          </div>
          <div className="margin-top-4">
            <div className="grid-row flex-first-baseline space-between-4">
              <Button
                appearance="outline"
                onClick={toggleModal}
                text="No, cancel"
              />
              <Button
                appearance="default"
                onClick={() => {
                  if (!monthStatus) {
                    setFilterByMonth(selectedMonth);
                    toggleModal();
                  }
                }}
                disabled={!!monthStatus}
                text="View roster"
              />
            </div>
          </div>
        </>
      }
    />
  );
};
