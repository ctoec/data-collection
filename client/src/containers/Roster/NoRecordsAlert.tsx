import { Alert } from '@ctoec/component-library';
import React from 'react';
import { AddRecordButton } from '../../components/AddRecordButton';
import { Organization } from '../../shared/models';

export const NoRecordsAlert: React.FC = () => (
  <Alert
    heading="No records in your roster"
    type="info"
    text="Get started by adding records to your roster"
    actionItem={<AddRecordButton id="add-record-in-alert" />}
  />
);
