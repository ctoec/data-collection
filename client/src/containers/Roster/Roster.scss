.roster {
  table-layout: fixed;
}

.bx--data-table th,
.bx--data-table td,
.bx--data-table tr:hover {
  background-color: #ffffff !important;
  color: #1b1b1b;
}

.bx--data-table thead tr {
  border-bottom: 2px solid #1b1b1b;
}

.bx--data-table td {
  line-height: 150%;
  white-space: break-spaces;
  padding-bottom: 2em;
  padding-top: 2em;
}

.action-button {
  width: fit-content !important;
  padding-left: 0 !important;
  padding-right: 0 !important;
}