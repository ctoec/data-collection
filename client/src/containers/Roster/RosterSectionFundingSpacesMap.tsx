import React from 'react';
import { Divider } from '@ctoec/component-library';
import { Child, FundingSource } from '../../shared/models';
import { getCurrentFunding } from '../../utils/models';
import { getStrippedFundingSourceName } from '../Home/utils/getFundingSpaceDisplayName';

type RosterSectionFundingSpacesMapProps = {
  children: Child[];
  hideCapacity: boolean;
};

type TimeAndCount = {
  time: string;
  count: number;
};

type SourceToTimes = {
  sourceName: string;
  capacity: number;
  times: TimeAndCount[];
};

/**
 * The "header" area for a roster section comprised of children of a
 * single age group. Contains information about capacity and filled
 * spaces of each funding group included in the children.
 */
export const RosterSectionFundingSpacesMap: React.FC<RosterSectionFundingSpacesMapProps> = ({
  children,
  hideCapacity,
}) => {
  const fundingSpacesMap: SourceToTimes[] = [];
  children.forEach((c) => {
    const fundingSpace = getCurrentFunding({ child: c })?.fundingSpace;
    if (fundingSpace) {
      let sourceMatch = fundingSpacesMap.find(
        (fs) => fs.sourceName === fundingSpace.source
      );
      if (!sourceMatch) {
        sourceMatch = {
          sourceName: fundingSpace.source,
          capacity: fundingSpace.capacity,
          times: [],
        } as SourceToTimes;
        fundingSpacesMap.push(sourceMatch);
      }
      let timeMatch = sourceMatch.times.find(
        (t) => t.time === fundingSpace.time
      );
      if (!timeMatch) {
        timeMatch = { time: fundingSpace.time, count: 0 } as TimeAndCount;
        sourceMatch.times.push(timeMatch);
      }
      timeMatch.count += 1;
    }
  });

  return (
    <div>
      <Divider />
      {fundingSpacesMap.map((sourceToTimes) => {
        let displayStr = (
          <>
            {sourceToTimes.times.map((t) => (
              <>
                <b>{t.time}</b> {t.count}
                {hideCapacity || sourceToTimes.capacity === -1
                  ? ''
                  : `/${sourceToTimes.capacity}`}{' '}
                &nbsp;&nbsp;
              </>
            ))}
          </>
        );
        return (
          <>
            <div key={sourceToTimes.sourceName} className="grid-row grid-gap">
              <div className="tablet:grid-col-3 font-body-sm text-bold margin-top-1 margin-left-2">
                {getStrippedFundingSourceName(
                  sourceToTimes.sourceName as FundingSource
                )}
              </div>
              <div className="tablet:grid-col-8 font-body-sm usa-hint margin-top-1">
                {displayStr}
              </div>
            </div>
            <div className="margin-top-1 margin-bottom-1">
              <Divider />
            </div>
          </>
        );
      })}
    </div>
  );
};
