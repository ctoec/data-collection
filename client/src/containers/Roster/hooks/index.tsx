export * from './useChildrenWithAlerts';
export * from './useOrgSiteProps';
export * from './useUpdateRosterParams';
export * from './usePaginatedChildData';
