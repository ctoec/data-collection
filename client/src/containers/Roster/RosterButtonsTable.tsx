import {
  Button,
  Calendar,
  History,
  TextWithIcon,
} from '@ctoec/component-library';
import moment from 'moment';
import React, { useContext, useState } from 'react';
import RosterContext from '../../contexts/RosterContext/RosterContext';
import { AddRecordButton } from '../../components/AddRecordButton';
import { CSVExcelDownloadButton } from '../../components/CSVExcelDownloadButton';
import { MonthFilterModal } from './MonthFilterModal';
import { QUERY_STRING_MONTH_FORMAT } from '../../containers/Roster/rosterUtils';

export const RosterButtonsTable: React.FC = () => {
  const { query, updateQueryMonth, updateQueryWithdrawn } = useContext(
    RosterContext
  );
  const [isModalOpen, setIsModalOpen] = useState(false);
  const toggleModal = () => setIsModalOpen((o) => !o);
  return (
    <table className="oec-table buttons-table margin-top-2 margin-bottom-4">
      <tbody>
        <tr>
          <th scope="row">Actions</th>
          <td>
            <div className="display-flex">
              <AddRecordButton
                className="margin-right-2"
                id="add-record-in-actions"
              />
              <CSVExcelDownloadButton fileType="csv" whichDownload="roster" />
            </div>
          </td>
        </tr>
        <tr>
          <th scope="row">Views</th>
          <td>
            <div className="display-flex">
              <Button
                text={<TextWithIcon Icon={History} text="Withdrawn records" />}
                appearance="unstyled"
                onClick={() => updateQueryWithdrawn(true)}
                className="margin-right-2"
              />
              <Button
                text={
                  <TextWithIcon
                    Icon={Calendar}
                    text="Roster for a past month"
                  />
                }
                appearance="unstyled"
                onClick={toggleModal}
              />
              <MonthFilterModal
                filterByMonth={
                  moment.utc(query.month, QUERY_STRING_MONTH_FORMAT) ??
                  undefined
                }
                setFilterByMonth={updateQueryMonth}
                toggleModal={toggleModal}
                isModalOpen={isModalOpen}
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  );
};
