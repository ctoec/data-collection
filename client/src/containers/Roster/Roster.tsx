import React, { useContext } from 'react';
import { Moment } from 'moment';
import {
  Accordion,
  Button,
  TabNav,
  HeadingLevel,
} from '@ctoec/component-library';
import { stringify, parse } from 'query-string';
import moment from 'moment';
import UserContext from '../../contexts/UserContext/UserContext';
import { FixedBottomBar } from '../../components/FixedBottomBar/FixedBottomBar';
import { getH1RefForTitle } from '../../utils/getH1RefForTitle';
import { useHistory, useLocation } from 'react-router-dom';
import { apiPut } from '../../utils/api';
import AuthenticationContext from '../../contexts/AuthenticationContext/AuthenticationContext';
import {
  getAccordionItems,
  getChildrenByAgeGroup,
  getQueryMonthFormat,
  QUERY_STRING_MONTH_FORMAT,
  applyClientSideFilters,
} from './rosterUtils';
import { BackButton } from '../../components/BackButton';
import { RosterButtonsTable } from './RosterButtonsTable';
import { MonthFilterIndicator } from './MonthFilter/MonthFilterIndicator';
import { Child } from '../../shared/models';
import { NoRecordsAlert } from './NoRecordsAlert';
import {
  useUpdateRosterParams,
  useGenerateUserSpecificProps,
  useChildrenWithErrorsAlert,
} from './hooks';
import { useAuthenticatedSWR } from '../../hooks/useAuthenticatedSWR';

const Roster: React.FC = () => {
  const h1Ref = getH1RefForTitle();
  const { user } = useContext(UserContext);
  const isSiteLevelUser = user?.accessType === 'site';
  const userOrganizations = user?.organizations || [];
  const isMultiOrgUser = userOrganizations.length > 1;

  const { accessToken } = useContext(AuthenticationContext);
  const history = useHistory();

  // Parse query params, and update if missing (i.e. initial load) or invalid
  const location = useLocation();
  const query = parse(location.search) as {
    organization?: string;
    site?: string;
    month?: string;
  };

  const queryMonth = query.month
    ? moment.utc(query.month, QUERY_STRING_MONTH_FORMAT)
    : undefined;
  useUpdateRosterParams();

  // Fetch child data, filtered by organization and month
  const { data: children, isValidating } = useAuthenticatedSWR<Child[]>(
    query.organization
      ? `/children?${stringify({ organizationId: query.organization })}`
      : null
  );

  // Organization filtering happens on the server-side,
  // but site filtering needs to happen in the client-side, if a
  // site is requested
  const browserSideFilteredChildren = applyClientSideFilters(
    children || [],
    query.site,
    queryMonth
  );
  const childrenByAgeGroup = getChildrenByAgeGroup(browserSideFilteredChildren);
  const accordionProps = {
    items: getAccordionItems(childrenByAgeGroup, {
      hideCapacity: isSiteLevelUser,
      showOrgInTables: isMultiOrgUser,
    }),
    titleHeadingLevel: 'h2' as HeadingLevel,
  };

  // Get alerts for page, including alert for children with errors
  const { alertElements } = useChildrenWithErrorsAlert(
    isValidating,
    browserSideFilteredChildren.filter(
      (child) => child.validationErrors && child.validationErrors.length
    ).length
  );
  // Get props for tabNav, h1Text, and subHeaderText based on user access (i.e. user's sites and org permissions)
  const { tabNavProps, h1Text, subHeaderText } = useGenerateUserSpecificProps(
    isValidating,
    browserSideFilteredChildren.length
  );

  // Function to submit data to OEC, to pass down into submit button
  async function submitToOEC() {
    // If there's an active org submit
    if (query.organization) {
      await apiPut(`oec-report/${query.organization}`, undefined, {
        accessToken,
      });
      history.push('/success');
    }
    // Otherwise, do nothing (button should be disabled)
  }

  // Function to update active month, to pass down into month filter buttons
  const updateActiveMonth = (newMonth: Moment) => {
    const month = getQueryMonthFormat(newMonth);
    history.push({
      search: stringify({
        organization: query.organization,
        site: query.site,
        month,
      }),
    });
  };

  return (
    <>
      <div className="Roster grid-container">
        <BackButton
          text="Back to getting started"
          location="/getting-started"
        />
        {alertElements}
        <div className="grid-row flex-align-center">
          <div className="tablet:grid-col-10">
            <h1 className="margin-bottom-0" ref={h1Ref}>
              {isSiteLevelUser && (
                <div className="margin-bottom-1 font-body-sm text-base-darker">
                  {userOrganizations[0].providerName}
                </div>
              )}
              {h1Text}
            </h1>
            <p className="font-body-xl margin-top-1">{subHeaderText}</p>
          </div>
          <div className="tablet:grid-col-2">
            <MonthFilterIndicator
              filterByMonth={queryMonth}
              setFilterByMonth={updateActiveMonth}
            />
          </div>
        </div>
        <RosterButtonsTable
          filterByMonth={queryMonth}
          setFilterByMonth={updateActiveMonth}
        />
        {isValidating ? (
          <></>
        ) : !(children || []).length ? (
          <NoRecordsAlert />
        ) : tabNavProps ? (
          <TabNav {...tabNavProps}>
            {accordionProps.items.length ? (
              <Accordion {...accordionProps} />
            ) : (
              <NoRecordsAlert />
            )}
          </TabNav>
        ) : accordionProps.items.length ? (
          <Accordion {...accordionProps} />
        ) : (
          <NoRecordsAlert />
        )}
      </div>

      <FixedBottomBar>
        <Button
          text="Back to getting started"
          href="/getting-started"
          appearance="outline"
        />
        {!isSiteLevelUser && (
          <Button
            text={
              isSiteLevelUser
                ? 'Organization permissions required to submit'
                : 'Send to OEC'
            }
            onClick={submitToOEC}
            disabled={!query.organization}
          />
        )}
      </FixedBottomBar>
    </>
  );
};

export default Roster;
