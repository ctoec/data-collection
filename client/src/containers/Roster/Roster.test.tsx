import React from 'react';
import { waitFor } from '@testing-library/dom';
import moment from 'moment';
import {
  snapshotTestHelper,
  accessibilityTestHelper,
  renderHelper,
} from '../../testHelpers';
import Roster from './Roster';
import UserContext from '../../contexts/UserContext/UserContext';
import {
  Family,
  Child,
  Site,
  FundingSpace,
  FundingSource,
  AgeGroup,
  ReportingPeriod,
  Funding,
  User,
} from '../../shared/models';

jest.mock('../../utils/api');
import * as api from '../../utils/api';
import DataCacheContext from '../../contexts/DataCacheContext/DataCacheContext';
const apiMock = api as jest.Mocked<typeof api>;

const _child = {
  firstName: '',
  lastName: '',
  id: '',
} as Child;

const INFANT_TODDLER_CDC = {
  id: 1,
  source: FundingSource.CDC,
  ageGroup: AgeGroup.InfantToddler,
  capacity: 3,
} as FundingSpace;

const INFANT_TODDLER_PSR = {
  id: 2,
  source: FundingSource.PSR,
  ageGroup: AgeGroup.InfantToddler,
  capacity: 5,
};

const PRESCHOOL_CDC = {
  id: 3,
  source: FundingSource.CDC,
  ageGroup: AgeGroup.Preschool,
  capacity: 10,
};

const children: Child[] = [
  {
    id: '0000000-0000-0000-0000-0000000000000',
    firstName: 'First',
    lastName: 'Toddler',
    birthdate: moment.utc('2019-06-01', 'YYYY-MM-DD'),
    family: {} as Family,
    enrollments: [
      {
        child: _child,
        id: 1,
        site: { id: 1, siteName: 'Site' } as Site,
        ageGroup: AgeGroup.InfantToddler,
        fundings: [
          {
            id: 1,
            fundingSpace: INFANT_TODDLER_CDC,
            firstReportingPeriod: {} as ReportingPeriod,
          } as Funding,
        ],
      },
    ],
  },
  {
    id: '11111111-0000-0000-0000-0000000000000',
    firstName: 'Second',
    lastName: 'Toddler',
    birthdate: moment.utc('2018-11-01', 'YYYY-MM-DD'),
    family: {} as Family,
    enrollments: [
      {
        child: _child,
        id: 1,
        site: { id: 1, siteName: 'Site' } as Site,
        ageGroup: AgeGroup.InfantToddler,
        fundings: [
          {
            id: 1,
            fundingSpace: INFANT_TODDLER_PSR,
            firstReportingPeriod: {} as ReportingPeriod,
          } as Funding,
        ],
      },
    ],
  } as Child,
  {
    id: '22222222-0000-0000-0000-0000000000000',
    firstName: 'First',
    lastName: 'Preschool',
    birthdate: moment.utc('2016-01-01', 'YYYY-MM-DD'),
    family: {} as Family,
    enrollments: [
      {
        child: _child,
        id: 1,
        site: { id: 1, siteName: 'Site' } as Site,
        ageGroup: AgeGroup.Preschool,
        fundings: [
          {
            id: 1,
            fundingSpace: PRESCHOOL_CDC,
            firstReportingPeriod: {} as ReportingPeriod,
          } as Funding,
        ],
      },
    ],
  },
];

const oneOrgUser = {
  organizations: [{ id: 1, providerName: 'Organization' }],
} as User;

const multiOrgUser = {
  organizations: [
    { id: 1, providerName: 'Organization' },
    { id: 2, providerName: 'Org 2' },
  ],
} as User;

describe('Roster', () => {
  const helperOpts = {
    wrapInRouter: true,
  };
  snapshotTestHelper(
    <UserContext.Provider value={{ user: oneOrgUser, loading: false }}>
      <DataCacheContext.Provider
        value={{
          children: {
            records: children,
            loading: false,
            addOrUpdateRecord: jest.fn(),
          },
        }}
      >
        <Roster />
      </DataCacheContext.Provider>
    </UserContext.Provider>,
    helperOpts,
    'matches snapshot for user with 1 org'
  );

  snapshotTestHelper(
    <UserContext.Provider value={{ user: multiOrgUser, loading: false }}>
      <DataCacheContext.Provider
        value={{
          children: {
            records: children,
            loading: false,
            addOrUpdateRecord: jest.fn(),
          },
        }}
      >
        <Roster />
      </DataCacheContext.Provider>
    </UserContext.Provider>,
    helperOpts,
    'matches snapshot for user with >1 org'
  );

  accessibilityTestHelper(
    <UserContext.Provider value={{ user: multiOrgUser, loading: false }}>
      <DataCacheContext.Provider
        value={{
          children: {
            records: children,
            loading: false,
            addOrUpdateRecord: jest.fn(),
          },
        }}
      >
        <Roster />
      </DataCacheContext.Provider>
    </UserContext.Provider>,
    helperOpts
  );

  it('correctly separates children by ageGroup', async () => {
    const renderResult = await renderHelper(
      <DataCacheContext.Provider
        value={{
          children: {
            records: children,
            loading: false,
            addOrUpdateRecord: jest.fn(),
          },
        }}
      >
        <Roster />
      </DataCacheContext.Provider>,
      helperOpts
    );

    // Assert there are two roster sections
    const accordionHeaders = (
      await renderResult.findAllByText(/\d+ child(ren)?$/)
    ).map((innerSpanElement) => innerSpanElement.parentElement as HTMLElement);
    expect(accordionHeaders).toHaveLength(2);

    // Assert first section (infant/toddler) has 2 items in table,
    const infantToddlerHeader = accordionHeaders.find((header) =>
      header.innerHTML.includes(AgeGroup.InfantToddler)
    );
    expect(infantToddlerHeader).toContainHTML('2 children');

    // and second (preschool) has 1
    const preschoolHeader = accordionHeaders.find((header) =>
      header.innerHTML.includes(AgeGroup.Preschool)
    );
    expect(preschoolHeader).toContainHTML('1 child');
  });

  afterAll(() => jest.clearAllMocks());
});
