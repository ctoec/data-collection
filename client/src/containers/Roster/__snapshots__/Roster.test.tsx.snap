// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roster matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid-container"
  >
    <h2
      class="font-body-xl margin-bottom-0"
    >
      Organization
    </h2>
    <p
      class="font-body-xl margin-top-1"
    >
      3 children enrolled at 1 sites
    </p>
    <div
      class="display-flex flex-col flex-align center flex-justify-start margin-top-2 margin-bottom-4"
    >
      <a
        class="usa-button usa-button--unstyled"
        href="/create-record"
      >
        <span
          class="oec-text-with-icon"
        >
          Add a record
        </span>
      </a>
      <button
        class="usa-button usa-button--unstyled margin-left-2"
        type="button"
      >
        <span
          class="oec-text-with-icon"
        >
          Export current view (PLACEHOLDER)
        </span>
      </button>
    </div>
    <div
      aria-multiselectable="true"
      class="oec-accordion"
    >
      <h2
        class="oec-accordion__heading"
      >
        <button
          aria-controls="Infant/toddler"
          aria-expanded="false"
          class="oec-accordion__button"
        >
          <div
            class="display-flex flex-justify flex-col flex-align-center padding-top-3 padding-bottom-3"
          >
            <div
              class="oec-accordion__heading-title"
            >
              <p>
                <span
                  class="text-bold"
                >
                  Infant/toddler
                </span>
                 2 children
              </p>
            </div>
            <div
              class="oec-accordion__heading-expand"
            >
              Hide Infant/toddler roster
            </div>
          </div>
        </button>
      </h2>
      <div
        aria-hidden="true"
        class="oec-accordion__content"
        id="Infant/toddler"
      >
        <table
          class="oec-table"
          id="roster-table-Infant/toddler"
        >
          <thead>
            <tr>
              <th
                aria-sort="none"
                class="oec-table__column-header oec-sortable oec-unsorted text-pre text-center font-body-2xs"
                role="columnheader"
                scope="col"
              >
                <button
                  aria-label="Sort table by Name in ascending order"
                  class="oec-table__column-title usa-button--unstyled width-full"
                >
                  Name
                  <div
                    class="oec-table__sort-controls"
                  />
                </button>
              </th>
              <th
                aria-sort="none"
                class="oec-table__column-header oec-unsorted text-pre text-center font-body-2xs"
                role="columnheader"
                scope="col"
              >
                <span
                  class="oec-table__column-title"
                >
                  Birth Date
                </span>
              </th>
              <th
                aria-sort="none"
                class="oec-table__column-header oec-unsorted text-pre text-center font-body-2xs"
                role="columnheader"
                scope="col"
              >
                <span
                  class="oec-table__column-title"
                >
                  Age group
                </span>
              </th>
              <th
                aria-sort="none"
                class="oec-table__column-header oec-unsorted text-pre text-center font-body-2xs"
                role="columnheader"
                scope="col"
              >
                <span
                  class="oec-table__column-title"
                >
                  Funding type
                </span>
              </th>
              <th
                aria-sort="none"
                class="oec-table__column-header oec-unsorted text-pre text-center font-body-2xs"
                role="columnheader"
                scope="col"
              >
                <span
                  class="oec-table__column-title"
                >
                  Contract space
                </span>
              </th>
              <th
                aria-sort="none"
                class="oec-table__column-header oec-unsorted text-pre text-center font-body-2xs"
                role="columnheader"
                scope="col"
              >
                <span
                  class="oec-table__column-title"
                >
                  Site
                </span>
              </th>
              <th
                aria-sort="none"
                class="oec-table__column-header oec-unsorted text-pre text-center font-body-2xs"
                role="columnheader"
                scope="col"
              >
                <span
                  class="oec-table__column-title"
                >
                  Enrollment date
                </span>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <a
                  href="/edit-record/0000000-0000-0000-0000-0000000000000"
                >
                  <button
                    class="usa-button usa-button--unstyled text-no-wrap font-body-2xs"
                    type="button"
                  >
                    First Toddler
                  </button>
                </a>
              </td>
              <td />
              <td>
                Infant/toddler
              </td>
              <td>
                CDC
              </td>
              <td />
              <td />
              <td />
            </tr>
            <tr>
              <td>
                <a
                  href="/edit-record/11111111-0000-0000-0000-0000000000000"
                >
                  <button
                    class="usa-button usa-button--unstyled text-no-wrap font-body-2xs"
                    type="button"
                  >
                    Second Toddler
                  </button>
                </a>
              </td>
              <td />
              <td>
                Infant/toddler
              </td>
              <td>
                PSR
              </td>
              <td />
              <td />
              <td />
            </tr>
          </tbody>
        </table>
      </div>
      <h2
        class="oec-accordion__heading"
      >
        <button
          aria-controls="Preschool"
          aria-expanded="false"
          class="oec-accordion__button"
        >
          <div
            class="display-flex flex-justify flex-col flex-align-center padding-top-3 padding-bottom-3"
          >
            <div
              class="oec-accordion__heading-title"
            >
              <p>
                <span
                  class="text-bold"
                >
                  Preschool
                </span>
                 1 children
              </p>
            </div>
            <div
              class="oec-accordion__heading-expand"
            >
              Hide Preschool roster
            </div>
          </div>
        </button>
      </h2>
      <div
        aria-hidden="true"
        class="oec-accordion__content"
        id="Preschool"
      >
        <table
          class="oec-table"
          id="roster-table-Preschool"
        >
          <thead>
            <tr>
              <th
                aria-sort="none"
                class="oec-table__column-header oec-sortable oec-unsorted text-pre text-center font-body-2xs"
                role="columnheader"
                scope="col"
              >
                <button
                  aria-label="Sort table by Name in ascending order"
                  class="oec-table__column-title usa-button--unstyled width-full"
                >
                  Name
                  <div
                    class="oec-table__sort-controls"
                  />
                </button>
              </th>
              <th
                aria-sort="none"
                class="oec-table__column-header oec-unsorted text-pre text-center font-body-2xs"
                role="columnheader"
                scope="col"
              >
                <span
                  class="oec-table__column-title"
                >
                  Birth Date
                </span>
              </th>
              <th
                aria-sort="none"
                class="oec-table__column-header oec-unsorted text-pre text-center font-body-2xs"
                role="columnheader"
                scope="col"
              >
                <span
                  class="oec-table__column-title"
                >
                  Age group
                </span>
              </th>
              <th
                aria-sort="none"
                class="oec-table__column-header oec-unsorted text-pre text-center font-body-2xs"
                role="columnheader"
                scope="col"
              >
                <span
                  class="oec-table__column-title"
                >
                  Funding type
                </span>
              </th>
              <th
                aria-sort="none"
                class="oec-table__column-header oec-unsorted text-pre text-center font-body-2xs"
                role="columnheader"
                scope="col"
              >
                <span
                  class="oec-table__column-title"
                >
                  Contract space
                </span>
              </th>
              <th
                aria-sort="none"
                class="oec-table__column-header oec-unsorted text-pre text-center font-body-2xs"
                role="columnheader"
                scope="col"
              >
                <span
                  class="oec-table__column-title"
                >
                  Site
                </span>
              </th>
              <th
                aria-sort="none"
                class="oec-table__column-header oec-unsorted text-pre text-center font-body-2xs"
                role="columnheader"
                scope="col"
              >
                <span
                  class="oec-table__column-title"
                >
                  Enrollment date
                </span>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <a
                  href="/edit-record/22222222-0000-0000-0000-0000000000000"
                >
                  <button
                    class="usa-button usa-button--unstyled text-no-wrap font-body-2xs"
                    type="button"
                  >
                    First Preschool
                  </button>
                </a>
              </td>
              <td />
              <td>
                Preschool
              </td>
              <td>
                CDC
              </td>
              <td />
              <td />
              <td />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
